fix: preview thumbnails don't scale as large as they could

Computerdores opened the pull request:
### Summary
Due to a hardcoded maximum size for preview thumbnails, preview thumbnails don't scale to fit all the space available to the PreviewThumb Component. Since a limit is needed to avoid the performance hit that massive images would incur, I have change the limit to be the screen size since the component shouldn't be bigger than that anyways.

### Tasks Completed
-   Platforms Tested:
    -   [x] Windows x86
    -   [ ] Windows ARM
    -   [ ] macOS x86
    -   [ ] macOS ARM
    -   [ ] Linux x86
    -   [ ] Linux ARM
-   Tested For:
    -   [x] Basic functionality
    -   [ ] PyInstaller executable

Computerdores commented:
Before:
<img width="1920" height="1040" alt="image" src="https://github.com/user-attachments/assets/328c5492-6908-4259-bde6-d83d556fe097" />
After
<img width="1920" height="1040" alt="image" src="https://github.com/user-attachments/assets/7d7cdfc9-d5a5-404b-b4e9-4ad6712810d3" />

CyanVoxel commented:
Nice fix! Although using the screen size can come with a decent performance hit depending on the screen resolution and underlying file type. For example, SVG files now take about a full second to render in the preview panel for me instead of showing instantly since they're now always rendering at 5K.

One alternative that gets around this problem could be to leverage the `self.__img_button_size` variable already in the class:
```py
def __render_thumb(self, filepath: Path) -> None:
        self.__thumb_renderer.render(
            time.time(),
            filepath,
            self.__img_button_size,
            self.devicePixelRatio(),
            update_on_ratio_change=True,
        )
```
This has the upside of only rendering what's needed for the preview panel's current size, but comes with the drawback of not automatically filling in the entire space available if the preview panel is made larger after rendering until re-rendering the preview. Perhaps some sort of threshold check inside the `resizeEvent` to trigger a re-render could fix that though?

Computerdores commented:
> Nice fix! Although using the screen size can come with a decent performance hit depending on the screen resolution and underlying file type. For example, SVG files now take about a full second to render in the preview panel for me instead of showing instantly since they're now always rendering at 5K.
> 
> One alternative that gets around this problem could be to leverage the `self.__img_button_size` variable already in the class:
> 
> ```python
> def __render_thumb(self, filepath: Path) -> None:
>         self.__thumb_renderer.render(
>             time.time(),
>             filepath,
>             self.__img_button_size,
>             self.devicePixelRatio(),
>             update_on_ratio_change=True,
>         )
> ```
> 
> This has the upside of only rendering what's needed for the preview panel's current size, but comes with the drawback of not automatically filling in the entire space available if the preview panel is made larger after rendering until re-rendering the preview. Perhaps some sort of threshold check inside the `resizeEvent` to trigger a re-render could fix that though?

Yeah... this pretty much sums up most of the thoughts I have had on this. Main reason why I did it as it is, is that this was the path of least resistance. Also, any other approach requires rerendering on resize and I am not quite sure how to do that well. Because you either rerender on every resize event (horrible performance) or you try to guess how big the user will make it and only resize when it gets above that (very guessy and kind of feels icky to me, not sure that's a good enough argument though 😅)

Computerdores commented:
> On my display it only re-renders up to one additional time when going from a small preview to a massive one.

I am assuming you pressed the full screen button of the window? Because for me when dragging the slider of the splitter, for example, it issued multiple resize events (in testing from earlier, not at my pc atm so can't test)

Computerdores commented:
what are you setting `self.__filepath` for? Couldn't you instead check whether `self.__rendered_res` is None?

Computerdores commented:
I implemented what you suggested, only change is that I moved the factor to a constant so it can be tweaked more easily

Computerdores commented:
> > I implemented what you suggested, only change is that I moved the factor to a constant so it can be tweaked more easily
> 
> Looks good! Probably just want to have `self.__filepath` and `self.__rendered_res` declared and/or initialized at the class level?

Damn I knew I missed something, fixed now
